File: day-1-sonar-sweep/src/main.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>> where P: AsRef<Path>, {
    let file = File::open(filename)?;
    Ok(io::BufReader::new(file).lines())
}

fn count_increased_depths() -> u32 {
    let mut previous_value: u32 = std::u32::MAX;
    let mut counter = 0;
    if let Ok(lines) = read_lines("input.txt") {
        for line in lines {
            if let Ok(actual_line) = line {
                let current_value: u32 = actual_line.parse().unwrap();
                if previous_value < current_value {
                    counter += 1;
                }
                previous_value = current_value;
            }
        }
    }
    return counter;
}

fn main() {
    println!("Counter of descents: {}", count_increased_depths());
}

File: day-1-sonar-sweep/python/main.py
import pathlib
import os

def main():
    print(count_increased_depths())

def count_increased_depths() -> int:
    counter: int = 0
    with open(os.path.join(pathlib.Path().resolve(), 'input.txt'), 'r') as reader:
        values: list[int] = list(map(lambda x: int(x), reader.readlines()))
        for x in range(1, len(values)):
            # print("{} - {}".format(x, values[x]))
            current_value = values[x]
            previous_value = values[x-1]
            if previous_value < current_value:
                counter += 1
    return counter

if __name__ == "__main__":
    main()